crates/indexer/src/http/metadata_json.rs

```diff
@@ -634,49 +634,75 @@ async fn dispatch_metadata_document(
+                    let (name, mint_address, collection_address) = metadatas::table
+                        .left_join(
+                            metadata_collection_keys::table.on(metadatas::mint_address
+                                .eq(metadata_collection_keys::collection_address)),
+                        )
+                        .filter(metadatas::address.eq(&addr))
+                        .select((
+                            metadatas::name,
+                            metadatas::mint_address,
+                            metadata_collection_keys::collection_address.nullable(),
+                        ))
+                        .first(db)
+                        .context("failed to load mint and name for search doc")?;
+
+                    let (creator_address, creator_twitter_handle) = metadata_creators::table
+                        .left_join(
+                            twitter_handle_name_services::table
+                                .on(metadata_creators::creator_address
+                                    .eq(twitter_handle_name_services::wallet_address)),
+                        )
+                        .filter(metadata_creators::metadata_address.eq(&addr))
+                        .filter(metadata_creators::verified.eq(true))
+                        .filter(metadata_creators::position.eq(0))
+                        .select((
+                            metadata_creators::creator_address,
+                            twitter_handle_name_services::twitter_handle.nullable(),
+                        ))
+                        .first(db)
+                        .context("failed to load creators for search document")?;
+
+                    Result::<_>::Ok((
+                        name,
+                        mint_address,
+                        collection_address,
+                        creator_address,
+                        creator_twitter_handle,
                     ))
-                    .first(db)
-                    .context("failed to load creators for search document")?;
-
-                Result::<_>::Ok((name, mint_address, creator_address, creator_twitter_handle))
-            }
-        })
-        .await
-        .map_err(|e| warn!("Failed to get search document data for metadata: {:?}", e))
+                }
+            })
+            .await
+            .map_err(|e| warn!("Failed to get search document data for metadata: {:?}", e))
     {
+        let document = MetadataDocument {
+            name,
+            mint_address,
+            image,
+            creator_address,
+            creator_twitter_handle,
+        };
+
         client
             .search()
-            .upsert_metadata(is_for_backfill, addr, MetadataDocument {
-                name,
-                mint_address,
-                image,
-                creator_address,
-                creator_twitter_handle,
-            })
+            .upsert_metadata(is_for_backfill, addr, document.clone())
```

Comment:
We can probably adjust the client API to borrow this rather than cloning it for better performance.